internal/pkg/cli/local_run.go

```diff
@@ -297,16 +301,52 @@ func (o *localRunOpts) Execute() error {
 	}
 	o.imageInfoList = append(o.imageInfoList, sidecarImageLocations...)
 
-	err = o.runPauseContainer(context.Background(), ports)
-	if err != nil {
-		return fmt.Errorf("run pause container: %w", err)
-	}
+	g, ctx := errgroup.WithContext(context.Background())
+	gotSigInt := &atomic.Bool{}
+	finishedErr := errors.New("containers stopped")
```

Comment:
Why do we define `finishedErr` as an error if it's always handled. Can't you just return `nil` instead of finishedErr and then handle other errors normally at the end of this function?

The only justification I can think of is that this is a potentially expected (though unwanted) behavior, as per Go's typical error handling. Why then in this case don't you at least log the error though?

Reply:
this is more to do with the interaction between `errgroup` and `ctx`.

If all of the containers exit on their own, the first goroutine will return `finishedErr`, which causes `errgroup` to cancel `ctx`, which the ctrl+c routine picks up on L334 so that it can exit itself.

Maybe this is clearer if I pass a separate context to the errgroup and cancel that context instead of returning this error to the errgroup?🤔

Reply:
Yeah I think that would make it more clear 👍